src/libappimage/core/AppImage.cpp

```diff
@@ -41,7 +41,7 @@ namespace appimage {
             format = getFormat(path);
 
             if (format == AppImageFormat::INVALID)
-                throw core::AppImageError("Unknown AppImage format");
+                throw core::AppImageError("Unknown AppImage format: "  << path);
```

Comment:
You can't use stream operators on arbitrary strings. Also, there's two spaces. Please just use `+`. If path is a `boost::filesystem::path`, use `path.string()`.

Reply:
Why does this work in Qt but not here? I am absolutely not a C++ guy...

Reply:
This also doesn't work in Qt that way. There may be more stream magic, e.g., Qt might have implemented a `<<` operator for their own `QString` class that works with left-hand arguments, but you're using `std::string` here.
Qt is not standard C++, it's often like a language put on top of C++. This here, however, is standard C++11.

Reply:
Which brings me to the point why we are using C++ and Boost rather than Qt? (Not saying we should switch immediately, but just curious.)

Reply:
This issue here is just about `std::string`, note the `std`. Boost is fast, convenient, mostly imported in the STL later on anyway (so when going for newer standards you can just `s/boost/std` in many cases), ...

Not using Qt allows non-Qt desktops to use this without political discussions.